### PR TITLE
first version of quick voice that starts inline chat with what you said

### PR DESCRIPTION
**Pull request by jrieken**
Things to figure out

* support DOWN and UP keybindings (for now I have used a chord)
* polish UI

### CODE REVIEW

File: src/vs/workbench/contrib/inlineChat/electron-sandbox/inlineChatQuickVoice.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+			switch (e.status) {
+				case SpeechToTextStatus.Started:
+					this._widget.active();
+					break;
+				case SpeechToTextStatus.Stopped:
+					break;
+				case SpeechToTextStatus.Recognizing:
+				case SpeechToTextStatus.Recognized:
+					message = e.text;
```
Comment (bpasero) [inline src/vs/workbench/contrib/inlineChat/electron-sandbox/inlineChatQuickVoice.ts:197]: Fyi there is a slight difference between the two:
- recognizing: will spit out words until pause
- recognized: will spit out the entire sequence of words from start or last pause to pause

I think the idea from the SDK is that a UI would show the former with a grey background (sort of unconfirmed) and the latter as final confirmed result. From my testing I have not seen a big difference in terms of quality, but here is how I apply it in my code:

https://github.com/microsoft/vscode/blob/86d1a94cca007f64e442c1aa0b6af60a40c1ceea/src/vs/workbench/contrib/chat/electron-sandbox/actions/voiceChatActions.ts#L286-L302

In other words: I append `Recognizing` text with whitespace until I receive `Recognized` where I drop all `Recognizing` text and append the result of `Recognized`

**Review by 22SIDHUXX — APPROVED** [review]

**Comment by jrieken** [comment]
@alexdima @ulugbekna the "interesting" commit for you keybinding folks is https://github.com/microsoft/vscode/pull/201314/commits/ca3a8724f4a3e6b53cac6426c4ea6f0c9239e207

**Review by alexdima — DISMISSED** [review]

File: src/vs/workbench/contrib/inlineChat/electron-sandbox/inlineChatActions.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+					listener.dispose(); // Event.once
+					if (listening) {
+						commandService.executeCommand(StopListeningAction.ID).finally(() => {
+							InlineChatController.get(editor)?.acceptInput();
+						});
+					} else {
+						handle.dispose();
+					}
+				}
+			);
```
Comment (ulugbekna) [inline src/vs/workbench/contrib/inlineChat/electron-sandbox/inlineChatActions.ts:78]: in future: I wonder if it would be better if the keybinding service fired an event that the holding-mode command registered for rather than setting up its own key_up DOM event listenter.

File: src/vs/workbench/services/keybinding/browser/keybindingService.ts [hunk 3]
```diff
@@ -269,6 +269,9 @@ export class WorkbenchKeybindingService extends AbstractKeybindingService {
 
 		// for standard keybindings
 		disposables.add(dom.addDisposableListener(window, dom.EventType.KEY_DOWN, (e: KeyboardEvent) => {
+			if (this._isInKeybindingHoldMode) {
+				return;
+			}
```
Comment (ulugbekna) [inline src/vs/workbench/services/keybinding/browser/keybindingService.ts:274]: If we start inline chat in listening mode and then close it clicking on x with the mouse, _isInKeybindingHoldMode is not set to true, breaking keybindings


https://github.com/microsoft/vscode/assets/16353531/c4c3b72c-d2f9-43c8-b057-d303eb98295c

File: src/vs/workbench/contrib/inlineChat/electron-sandbox/inlineChatActions.ts (hunk 2 above)
Comment (jrieken) [reply src/vs/workbench/contrib/inlineChat/electron-sandbox/inlineChatActions.ts:78]: Yeah, I was thinking to make enableHoldMode return a promise that resolves when hold more is over

**Review by ulugbekna — DISMISSED** [review]

**Review by ulugbekna — APPROVED** [review]
